src/EscapeAnalysis.jl

```diff
@@ -338,17 +364,22 @@ function find_escapes(ir::IRCode, nargs::Int)
                     escape_invoke!(stmt.args, pc, state, ir, changes)
                 elseif head === :new
                     info = state.ssavalues[pc]
-                    info == NotAnalyzed() && (info = NoEscape())
+                    if info == NotAnalyzed()
+                        info = NoEscape()
+                        push!(changes, (SSAValue(pc), info)) # we will later be interested in this allocation
+                    end
                     for arg in stmt.args[2:end]
                         push!(changes, (arg, info))
                     end
-                    push!(changes, (SSAValue(pc), info)) # we will be interested in if this allocation is not escape or not
+                    for (arg, finfo) in zip(stmt.args[2:end], info.FieldEscapes)
```

Comment:
`zip` has to allocate a new Tuple type for each type in the user's code (lots of dispatch), so we avoid that in the Core.Compiler

Reply:
It looks like there is not much difference ?
```julia
@eval Core.Compiler begin
    function tryzip(changes, ary, bry)
        for (a, b) in zip(ary, bry)
            push!(changes, (a, b))
        end
        return changes
    end
    function tryfor(changes, ary, bry)
        for i in 1:min(length(ary),length(bry))
            push!(changes, (ary[i], bry[i]))
        end
        return changes
    end
end

using BenchmarkTools
ary = collect(1:100);
bry = Char.(1:100);
@benchmark Core.Compiler.tryzip(changes, $ary, $bry) setup = (changes = Tuple{Int,Char}[])
@benchmark Core.Compiler.tryfor(changes, $ary, $bry) setup = (changes = Tuple{Int,Char}[])
```

I found `@inbounds` can speed up `tryfor` a bit, but it seems like it doesn't work well in `Core.Compiler` scope.

Reply:
The problem is not the values are new, it is that the type of `a` and `b` are new, so we must be careful to avoid putting the user's expressions into tuples.

Reply:
So this code snippet below is even problematic ?
```julia
n = min(length(args), length(info.FieldEscapes))
for i in 1:n
    arg = args[i]
    finfo = info.FieldEscapes[i]
    push!(changes, (arg, finfo))
end
```
, where `args` is arguments of user expressions ?

Then, I think we want to change the type of `changes` to `Vector{Pair{Any,EscapeLattice}}` from `Vector{Tuple{Any,EscapeLattice}}` ?

Reply:
Or perhaps `Union{SSAValue,Slot}`

Reply:
Addressed on master.